sdv/evaluation/single_table.py

```diff
@@ -27,7 +28,14 @@ def evaluate_quality(real_data, synthetic_data, metadata, verbose=True):
             Single table quality report object.
     """
     quality_report = QualityReport()
-    quality_report.generate(real_data, synthetic_data, metadata.to_dict(), verbose)
+    if not isinstance(metadata, Metadata):
+        metadata = Metadata()._convert_to_unified_metadata(metadata)
```

Comment:
This method converts the old metadata to a dict under the hood, then to the new metadata object. Then you convert it back to a dict below. Why not just convert it to a dict here and use that?